activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -273,10 +273,22 @@ def respond_to_missing?(name, include_private = false)
       def method_missing(method, *args, &block)
         if #{target}.respond_to?(method)
           #{target}.public_send(method, *args, &block)
-        elsif #{target}.nil?
-          raise DelegationError, "\#{method} delegated to #{target}, but #{target} is nil"
         else
-          super
+          delegation_error_if_target_nil(#{target}, method) do
+            super
+          end
+        end
+      end
+
+      def delegation_error_if_target_nil(target, method)
```

Comment:
This method should be private, and starts with `_` to avoid conflicts with the methods in the including class.